firebase_admin/project_management.py

```diff
@@ -214,49 +482,120 @@ def __init__(self, app):
-        """Retrieves detailed information about an Android App."""
+        return self._get_app_metadata(
+            platform_resource_name=_ProjectManagementService.ANDROID_APPS_RESOURCE_NAME,
+            identifier_name=_ProjectManagementService.ANDROID_APP_IDENTIFIER_NAME,
+            metadata_class=AndroidAppMetadata,
+            app_id=app_id)
+
+    def get_ios_app_metadata(self, app_id):
+        return self._get_app_metadata(
+            platform_resource_name=_ProjectManagementService.IOS_APPS_RESOURCE_NAME,
+            identifier_name=_ProjectManagementService.IOS_APP_IDENTIFIER_NAME,
+            metadata_class=IosAppMetadata,
+            app_id=app_id)
+
+    def _get_app_metadata(self, platform_resource_name, identifier_name, metadata_class, app_id):
+        """Retrieves detailed information about an Android or iOS app."""
         _check_is_nonempty_string(app_id, 'app_id')
-        path = '/v1beta1/projects/-/androidApps/{0}'.format(app_id)
+        path = '/v1beta1/projects/-/{0}/{1}'.format(platform_resource_name, app_id)
         response = self._make_request('get', path, app_id, 'App ID')
-        return AndroidAppMetadata(
+        return metadata_class(
+            response[identifier_name],
             name=response['name'],
             app_id=response['appId'],
-            display_name=response['displayName'],
-            project_id=response['projectId'],
-            package_name=response['packageName'])
+            display_name=response.get('displayName') or '',
+            project_id=response['projectId'])
+
+    def set_android_app_display_name(self, app_id, new_display_name):
+        self._set_display_name(
+            app_id=app_id,
+            new_display_name=new_display_name,
+            platform_resource_name=_ProjectManagementService.ANDROID_APPS_RESOURCE_NAME)
+
+    def set_ios_app_display_name(self, app_id, new_display_name):
+        self._set_display_name(
+            app_id=app_id,
+            new_display_name=new_display_name,
+            platform_resource_name=_ProjectManagementService.IOS_APPS_RESOURCE_NAME)
+
+    def _set_display_name(self, app_id, new_display_name, platform_resource_name):
+        """Sets the display name of an Android or iOS app."""
+        path = '/v1beta1/projects/-/{0}/{1}?updateMask=displayName'.format(
+            platform_resource_name, app_id)
+        request_body = {'displayName': new_display_name}
+        self._make_request('patch', path, app_id, 'App ID', json=request_body)
 
     def list_android_apps(self):
-        """Lists all the Android Apps within the Firebase Project."""
-        path = '/v1beta1/projects/{0}/androidApps?pageSize={1}'.format(
-            self._project_id, _ProjectManagementService.MAXIMUM_LIST_APPS_PAGE_SIZE)
+        return self._list_apps(
+            platform_resource_name=_ProjectManagementService.ANDROID_APPS_RESOURCE_NAME,
+            app_class=AndroidApp)
+
+    def list_ios_apps(self):
```

Comment:
What if we drop these platform-specific helper functions and directly call `_list_apps()` where it's needed? Same for others?

Reply:
This is to avoid the clutter that would otherwise appear at the module level. This way, I am able to localize the long list of platform-specific constants into the `_ProjectManagementService` class.